server/honeytrap.go

```diff
@@ -223,21 +223,21 @@ func (hc *Honeytrap) findService(conn net.Conn) *ServiceMap {
 			_n, err := pConn.Peek(buffer)
 			n = _n // avoid silly "variable not used" warning
 			if err == io.EOF {
-				return nil
+				return nil, nil
 			} else if err != nil {
 				log.Errorf(color.RedString("Could not peek bytes: %s", err.Error()))
-				return nil
+				return nil, nil
 			}
 			peekUninitialized = false
 		}
 		if ch.CanHandle(buffer[:n]) {
 			// Service supports payload
 			// TODO(capacitorset): pass peeked connection rather than the actual one
-			return service
+			return service, pConn
 		}
 	}
 	// No service can handle the connection. Let the caller deal with it.
-	return nil
+	return nil, nil
```

Comment:
shouldn't it return the original conn?

Reply:
What would be the point in returning the original connection?